turtleFSI/modules/newtonsolver.py

```diff
@@ -64,20 +64,26 @@ def newtonsolver(F, J_nonlinear, A_pre, A, b, bcs, lmbda, recompute, recompute_t
         if recompute_for_timestep or recompute_frequency or recompute_residual or recompute_initialize:
             if MPI.rank(MPI.comm_world) == 0 and verbose:
                 print("Compute Jacobian matrix")
+            # Assemble non-linear part of Jacobian, keep sparsity pattern (keep_diagonal=True)
             assemble(J_nonlinear, tensor=A,
                          form_compiler_parameters=compiler_parameters,
                          keep_diagonal=True)
+            # Add non-linear and linear part of Jacobian
             A.axpy(1.0, A_pre, True)
```

Comment:
```suggestion
            A.axpy(1.0, A_pre, True)
```
Note that @hkjeldsberg and I sat down and optimized some `axpy`'s in OasisMove yesterday. We can make them an order of magnitude faster (if last argument here is not `True`, which might happen in some cases).

Reply:
I will check with Henrik.